### PR TITLE
Add additional shader tutorials

### PR DESCRIPTION
**Pull request by user2332533**
These examples cover adding inputs (uniforms and buffers) to a shader. 

This follows on from the help provided in issue #1453 

These examples cover:
- A very basic vertex/frament shader
- Passing a number as a uniform
- Accessing multiple textures from a shader
- rendering to a new frambuffer texture

Only one additional housekeeping change was made:
- Moves gpu_particle_burst into the shaders submenu 

Tested locally, build succeeded and docs look fine.

### CODE REVIEW

**Comment by pushfoo** [comment]
Small nitpick: I'd rephrase the PR title as "Add additional shader tutorials" since the examples and tutorials folders are separate, and this PR adds content to the tutorials.

File: doc/tutorials/shader_inputs/index.rst [hunk 1]
```diff
@@ -0,0 +1,70 @@
+Working With Shaders
+================================
+
+Shaders are graphics programs that run on GPU and can be used for many varied purposes. 
+
+Here we look at some very simple shader programs and learn how to pass data to and from shaders
+
+Basic Pygame Program
```
Comment (pvcraven) [inline doc/tutorials/shader_inputs/index.rst]: Arcade, not Pygame

Comment (user2332533) [reply doc/tutorials/shader_inputs/index.rst]: Ooops, corrected in 9b8614

**Comment by csd4ni3l** [comment]
hmmmmm "Basic Pygame Program" in the "add shader examples to docs" commit in the index.rst file :laughing:

**Comment by user2332533** [comment]
Hopefully now with 100% fewer mistaken names. 

🤦‍♂️

File: doc/tutorials/shader_inputs/index.rst [hunk 2]
```diff
@@ -0,0 +1,70 @@
+Working With Shaders
+================================
```
Comment (pvcraven) [inline doc/tutorials/shader_inputs/index.rst]: Line too long, Sphinx will complain if it doesn't match the header.

File: doc/tutorials/shader_inputs/index.rst [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+.. literalinclude:: basic_shader.py
+    :caption: Simple shader
+    :linenos:
+    :emphasize-lines: 14-16, 18-37, 49-52
+
+Passing Data To The Shader
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To pass data to the shader program we can define `uniforms <https://www.khronos.org/opengl/wiki/Uniform_(GLSL)>`_. Uniforms are global shader variables that act as parameters passed from outside the shader program.
```
Comment (pvcraven) [inline doc/tutorials/shader_inputs/index.rst]: I know not everyone follows this, but I like keeping about an 80 char max on line-length for text.

**Comment by pvcraven** [comment]
I left a couple comments which were just simple formatting type things. Are you able to correct? 

This seems like a worth-while addition to docs.

**Comment by user2332533** [comment]
Hi Paul,

Sorry if I'm being dense, but I cannot find your comments for formatting etc. I'm happy to fix any issues but don't know where I should be looking for the comments.

**Comment by csd4ni3l** [comment]
just look at the changes of your commits(click on the commit) and it will appear there

**Comment by user2332533** [comment]
I've gone through all of the commits on this PR and the only comment I can see was Paul's Observation about the wrong library name, I corrected this in the second commit [9b86b14](https://github.com/pythonarcade/arcade/pull/1455/commits/9b86b14bb808a464b0a190357b2a2cc4e4d13eb8) and marked as resolved.

**Review by pvcraven — CHANGES_REQUESTED** [review]

**Comment by pvcraven** [comment]
> I've gone through all of the commits on this PR and the only comment I can see was Paul's Observation about the wrong library name, I corrected this in the second commit [9b86b14](https://github.com/pythonarcade/arcade/pull/1455/commits/9b86b14bb808a464b0a190357b2a2cc4e4d13eb8) and marked as resolved.

I didn't hit the 'finish review' button.

**Review by pvcraven — APPROVED** [review]

**Comment by pvcraven** [comment]
Will need to look at why this test is failing, don't think it has anything to do with the PR.

**Comment by pvcraven** [comment]
> Will need to look at why this test is failing, don't think it has anything to do with the PR

Currently failing because of an unrelated issue around a test and pyglet's 2.0.4 release. That'll need to get cleaned up separate from this.
